vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -143,17 +143,18 @@ def parse_pod(pod)
 
       # TODO, map volumes
       # TODO, podIP
+      containers_index = {}
       containers = pod.spec.containers
-      new_result[:container_definitions] = containers.collect do |container_def|
-        parse_container_definition(container_def, pod.metadata.uid)
-      end
-
-      # container instances
       unless pod.status.nil? || pod.status.containerStatuses.nil?
         pod.status.containerStatuses.each do |cn|
-          new_result[:containers] << parse_container(cn, pod.metadata.uid)
+          containers_index[cn.name] = parse_container(cn, pod.metadata.uid)
         end
       end
+      new_result[:container_definitions] = containers.collect do |container_def|
+        definition = parse_container_definition(container_def, pod.metadata.uid)
+        definition[:container] = containers_index[container_def.name]
+        definition
```

Comment:
This part can be improved as:

      new_result[:container_definitions] = containers.collect do |container_def|
        parse_container_definition(container_def, pod.metadata.uid).merge({
            :container => containers_index[container_def.name]
        })
      end